src/library_dylink.js

```diff
@@ -4,7 +4,7 @@
 // ==========================================================================
 
 var LibraryDylink = {
-#if MAIN_MODULE == 0
+#if !RELOCATABLE
```

Comment:
the corresponding comment on line 23 is now stale, and the one at the very bottom.

Reply:
also, while this does work, is it clearer? only a main module will use this, so that feels clearer to me the old way actually.

Reply:
The code in preable.js that calls `relocateExports` is currently guarded by `RELOCATABLE` not `MAIN_MODULE`.

I guess we need to think about what it might mean to build and run a program that is linked a `RELOCATABLE` but not linked with `MAIN_MODULE`?   Does that even make sense?   A lot of the code here in `library_dylink` relates to loading and relocating the wasm exports.

Perhaps I'm missing use case: What kind of use cases are there for programs that are linked with RELOCATABLE?  Is that even something we want to continue to expose to the user?

Reply:
I wouldn't wait on this for merging. But historically we did support relocatable programs that are not main modules for things like being able to adjust the region of memory used at runtime. So you can have a single big Memory with multiple independent modules, all relocatable, using their own "slice". And they can share pointers and data. But, not sure we want to support that...

Reply:
OK that makes some sense.  I've split I've preserved the original MAIN_MODULE check.   

I *think* it still makes sense for this function to live here along with all the other code relating to load and relocating code.     I guess you can think of this library as "dynamic linking and loading support"

PTAL.